Tests/SwiftParserTest/ExpressionTests.swift

```diff
@@ -897,3 +897,154 @@ final class MemberExprTests: XCTestCase {
     }
   }
 }
+
+final class StatementExpressionTests: XCTestCase {
+  func testIfExprInCoercion() {
+    AssertParse(
+      """
+      func foo() {
```

Comment:
I don’t think you actually need to have wrap the expression in a function declaration in all these test cases. Top-level should also allow expressions.